packages/runtime/container-runtime/src/compatUtils.ts

```diff
@@ -0,0 +1,159 @@
+		// constraint we need to have it defined, so we trick the type checker here.
+		"1.0.0": undefined as unknown as "on" | "delayed",
+		// We do not yet want to enable idCompressor by default since it will increase bundle sizes,
+		// and not all customers will benefit from it. Therefore, we will require customers to explicitly
+		// enable it. We are keeping it as a version-dependent option as this may change in the future.
+	},
+	explicitSchemaControl: {
+		"1.0.0": false,
+		// This option is unique since it was actually introduced before 2.0.0, but its purpose is to prevent 1.x clients from
+		// joining a session. Therefore, we will have it be `true` when the compatibility mode is set to >=2.0.0 and we do not
+		// want any 1.x clients to join. However, since this was not part of the default configurations before 3.0.0, we will
+		// keep it false unless a customer uses a compatibilityMode of 2.0.1 or higher.
+		"2.0.1": true,
+	},
+	flushMode: {
+		// Note: 1.x clients are compatible with TurnBased flushing, but here we elect to remain on Immediate flush mode
+		// as a work-around for inability to send batches larger than 1Mb. Immediate flushing keeps batches smaller as
+		// fewer messages will be included per flush.
+		"1.0.0": FlushMode.Immediate,
+		"2.0.0": FlushMode.TurnBased,
+	},
+	gcOptions: {
+		"1.0.0": {},
+		// Explicitly disable running Sweep in compat mode "2". Although sweep is supported in 2.x, it is disabled by default.
+		// This setting explicitly disables it to be extra safe.
+		"3.0.0": { enableGCSweep: true },
+	},
+};
+
+/**
+ * Returns the default version-dependent configs for a given compatibility mode.
+ */
+export function getConfigsForCompatMode<
+	T extends Record<string, unknown> = IContainerRuntimeOptionsVersionDependent,
+>(
+	compatibilityMode: SemanticVersion,
+	configMap: ConfigMap<T> = versionDependentOptionConfigMap as ConfigMap<T>,
+): T {
+	const defaultConfigs = {};
+	// Iterate over versionDependentOptionConfigMap to get default values for each version-dependent option.
+	for (const key of Object.keys(configMap)) {
+		const config = configMap[key as keyof T];
+		// Sort the versions in ascending order so we can short circuit the loop.
+		const versions = Object.keys(config).sort((a, b) => (semverGte(b, a) ? -1 : 1));
+		// For each config, we iterate over the keys and check if compatibilityMode is greater than or equal to the version.
+		// If so, we set it as the default value for the option. At the end of the loop we should have the most recent default
+		// value that is compatible with the version specified as the compatibilityMode.
+		for (const version of versions) {
+			if (semverGte(compatibilityMode, version)) {
+				defaultConfigs[key] = config[version as SemanticVersion];
+			} else {
+				// If the compatibility mode is less than the version, we break out of the loop since we don't need to check
```

Comment:
What happens if`compatibilityMode` is less than the least one present in `versionDependentOptionConfigMap`? It looks like it will return an empty set of configs. Is that what we want? Or should everything be disabled?